sktime/tests/test_all_estimators.py

```diff
@@ -54,6 +55,61 @@
 )
 
 
+generator_dict = dict()
+
+
+def is_excluded(test_name, est):
+    """Shorthand to check whether test test_name is excluded for estimator est."""
+    return test_name in EXCLUDED_TESTS.get(est.__name__, [])
+
+
+def _generate_estimator_class(test_name, **kwargs):
```

Comment:
It seems what this function does is filtering the estimators from ALL_ESTIMATORS, but the name suggests something else. How about renaming it to be more accurate? maybe `filter_estimator_classes`

Reply:
I disagree, since this is a "generic" following a template pattern, where functions are called `_generate_[fixturename]`.
The function happens to filter, yes, but it is concrete behaviour that fills an abstract hole whose name is "generate the fixtures of name `estimator_class`".

Reply:
the name describes the *purpose* of the function (that it generates fixtures) rather than its *logic* (i.e., how the fixture list is obtained). In this place, the purpose is more natural to use as a description.

Reply:
I now added detailed comments above the functions, which should further clarify things and explain the names better